src/plugins/analysis/architecture_detection/internal/dt.py

```diff
@@ -0,0 +1,78 @@
+import shlex
+import subprocess
+from subprocess import DEVNULL, PIPE
+from typing import Union
+
+import yaml
+
+
+def _get_compatible_entry(dts: str) -> Union[str, None]:
+    """
+    Returns the node name of /cpus/cpu*/compatible and its value from a device tree.
+    May return None because the spec does not guarantee the existence of this node.
+
+    See the DeviceTree spec for more information https://www.devicetree.org/specifications/
+    """
+
+    dtc_process = subprocess.run(
```

Comment:
I decided to just strip away huge propertys. This seems more robust since binary data can have many forms.
E.g.  `<0xd00dfeed 0x14908 0x38 0x126c8 0x28 0x11 .... >` and `[ff ff ff ...]`.
Skimming through the device tree spec I didn't find anything about binary data so there might be more ways to represent it.